### PR TITLE
web-server: add graceful shutdown

### PR DESCRIPTION
**Pull request by otaviopace**
## Description

Every time the server receives from the OS one of these
signals (SIGHUP, SIGINT, SIGTERM) it will stop receiving
requests and it will finish to serve all requests that were being
processes at the moment. After all that, the process will stop as
the OS requested.

## Your checklist for this pull request

:rotating_light: Please review this items for a good pull request. :four_leaf_clover:

1. I've read the project's [Contributing Guidelines](../CONTRIBUTING.md)
1. My commits are well written and follow [pagarme/git-style-guide](https://github.com/pagarme/git-style-guide)
1. My changes are well covered by **tests** and **logs**
1. I've updated the project docs (if needed)
1. I fell safe about this implementation
1. I feel comfortable with the code I wrote, and I'm not ashamed to show it to my friends

In a good pull request, everything above is true :relaxed:

this closes https://github.com/pagarme/ghostbusters/issues/33

### CODE REVIEW

File: src/graceful-shutdown/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const { forEachObjIndexed } = require('ramda')
```
Comment (grvcoelho) [inline src/graceful-shutdown/index.js]: Não sei se faz sentido essa pasta.

O que acha de criar uma pasta `src/server`? Lá a gente poderia ter dois arquivos: 

* `index.js` que contem o server express (acho que a gente deveria tirar lá do bin, e deixar lá só o mínimo necessário).
* `shutdown.js` onde a gente poderia guardar estas funções de shutdown e graceful reload.

File: src/graceful-shutdown/index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+const shutdown = (process, server, value) => () =>
+  server.close(() => process.exit(128 + value))
```
[the remark this answers is not in the document]
Comment (otaviopace) [reply src/graceful-shutdown/index.js]: http://tldp.org/LDP/abs/html/exitcodes.html

File: src/graceful-shutdown/index.js (hunk 1 above)
Comment (otaviopace) [reply src/graceful-shutdown/index.js]: hmm, gostei da ideia

File: src/server/index.js [hunk 3]
```diff
@@ -28,7 +29,9 @@ app.all('/boletos', defaultHandler)
 
 app.all('*', defaultResourceHandler)
 
-app.listen(
+const server = app.listen(
   PORT,
   () => console.log(`Superbowleto listening on port: ${PORT}`)
 )
+
+setupGracefulShutdown(process, expressServer)
```
Comment (grvcoelho) [inline src/server/index.js]: esqueceu de mudar aqui para `server`

File: src/server/index.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ const {
   defaultHandler,
```
Comment (grvcoelho) [inline src/server/index.js]: Você precisa mudar o entrypoint do docker-compose.yml

**Pull request merged**